### PR TITLE
Renamed and corrected Link Target Scanner

### PR DESCRIPTION
**Pull request by psiinon**
The vulnerability is on all 'target's that open a new window, not just _blank.
Also added support for a trusted domain config value - this isnt actually registered yet as plan to do tha via one of the release ones that will use the same key.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/pscanrulesAlpha/resources/Messages.properties [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-pscanalpha.blanktarget.refs=https://dev.to/ben/the-targetblank-vulnerability-by-example\nhttps://mathiasbynens.github.io/rel-noopener/\nhttps://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c\n
-pscanalpha.blanktarget.soln=Do not use a target of "_blank", or if you have to then also add the attribute: rel="noopener".
+pscanalpha.linktarget.name=Reverse tabnabbing
+pscanalpha.linktarget.desc=At least one link on this page is vulnerable to Reverse tabnabbing as it uses a target attribute without using both of the "noopener" and "noreferrer" keywords in the "rel" attribute, which allows the target page to take control of this page.
+pscanalpha.linktarget.refs=https://www.owasp.org/index.php/Reverse_Tabnabbing\nhttps://dev.to/ben/the-targetblank-vulnerability-by-example\nhttps://mathiasbynens.github.io/rel-noopener/\nhttps://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c\n
+pscanalpha.linktarget.soln=Do not use a target attribute, or if you have to then also add the attribute: rel="noopener noreferrer".
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/pscanrulesAlpha/resources/Messages.properties]: Kinda nit picky, but these should be moved alphabetically within the file as well ;)

File: src/org/zaproxy/zap/extension/pscanrulesAlpha/resources/help/contents/pscanalpha.html [hunk 2]
```diff
@@ -120,6 +116,13 @@ <H2>HTTP Server Response Header Scanner</H2>
 This checks response headers for the presence of a server header that contains version details.
 At LOW Threshold will raise an alert based on presence of the header field whether or not a version string is detected.
 
+<H2>Reverse tabnabbing</H2>
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/pscanrulesAlpha/resources/help/contents/pscanalpha.html]: Cap on T in Tabnabbing :+1:

**Comment by psiinon** [comment]
Hopefully all fixed :)

**Comment by thc202** [comment]
Thanks!

File: test/org/zaproxy/zap/extension/pscanrulesAlpha/LinkTargetScannerUnitTest.java [hunk 3]
```diff
@@ -428,4 +498,23 @@ public void shouldNotFailIfLinkOrAreaDoesNotHaveHref() throws Exception {
...
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        rule.getConfig().setProperty(LinkTargetScanner.TRUSTED_DOMAINS_PROPERTY, "[");
+        // When
+        msg.setResponseBody("<html><a href=\"https://www.example2.com/page1\" target=\"_blank\">link</a></html>");
+        msg.setResponseHeader(getHeader(HTML_CONTENT_TYPE, msg.getResponseBody().length()));
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+        // Then
+        // Then = no exception.
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrulesAlpha/LinkTargetScannerUnitTest.java]: Remove one of these?

**Comment by psiinon** [comment]
Yeah, just noticed that too ;)
Updated.

**Pull request merged**